tests/Aspire.Hosting.Testing.Tests/TestingFactoryTests.cs

```diff
@@ -28,31 +29,36 @@ public async Task HasEndPoints()
         Assert.True(pgConnectionString.Length > 0);
     }
 
-    [LocalOnlyFact]
+    [Fact]
+    [RequiresDocker]
     public void CanGetResources()
     {
         var appModel = _app.Services.GetRequiredService<DistributedApplicationModel>();
         Assert.Contains(appModel.GetContainerResources(), c => c.Name == "redis1");
         Assert.Contains(appModel.GetProjectResources(), p => p.Name == "myworker1");
     }
 
-    [LocalOnlyFact]
+    [Fact]
+    [RequiresDocker]
     public async Task HttpClientGetTest()
     {
         var httpClient = _app.CreateHttpClient("mywebapp1");
+        await Task.Delay(3000);
```

Comment:
@ReubenBond @DamianEdwards This PR enables the hosting.testing tests to run on CI. But two tests fail - `HttpClientGetTest`, and `SelectsFirstLaunchProfile` with connection refused. And it seems to be due to webapp not being ready. I added a `Task.delay` to confirm that, and it does work. But what is the correct way to have the test run only once the webapp is ready?

Reply:
You should configure the `HttpClient` with resilience so that it retries. Once #4444 goes in you'll be able to at least wait for the resource to transition to the "Running" state before attempting to send the request, but you still need the retries.

Reply:
Should that be added to the `Aspire.Hosting.Testing` api? I added that as an extension method in the test right now.

Reply:
`HttpClient` default resiliency pipelines aren't composable so we don't add it by default as there'd be no way to remove or modify it. The test templates have code in them that adds resiliency in the test itself.

Reply:
Look at the template, I added this to the `TestingAppHost1.AppHost/Program.cs`:
```csharp
builder.Services.ConfigureHttpClientDefaults(http =>
{
    http.AddStandardResilienceHandler();
});
```